Template sync

gq1 opened the pull request:
This workflow only can be triggered manually. That is why I need to merge to the main branch first, and then I can test.

Did two tests now and it seems working.
https://github.com/sanger-tol/genomenote/pull/179
https://github.com/sanger-tol/variantcalling/pull/110

I also tried to run all our pipelines without PRs:
https://github.com/sanger-tol/pipelines-website/actions/runs/14245925934

I think we can run some more steps to customize ToL changes

gq1 commented:
@sanger-tolsoft fix linting

muffato commented:
- Does it need to ask about self-hosted vs ubuntu-latest runners ? I can't see where it's used in the rest of the file
- We don't have a test pipeline (unlike https://github.com/nf-core/testpipeline). I don't think we need one ? I think we can assume that the template has been tested enough by the time we do it. So perhaps remove the option ?

I'm still trying to get it to run on blobtoolkit (already on the latest template so no PR should be created - but it shouldn't fail) and then will want to test on readmapping (on a recent template so the PR should be small / straightforward).

gq1 commented:
> * Does it need to ask about self-hosted vs ubuntu-latest runners ?
Removed.

> * We don't have a test pipeline (unlike https://github.com/nf-core/testpipeline). I don't think we need one ? I think we can assume that the template has been tested enough by the time we do it. So perhaps remove the option ?
Removed.

> 
> I'm still trying to get it to run on blobtoolkit (already on the latest template so no PR should be created - but it shouldn't fail) and then will want to test on readmapping (on a recent template so the PR should be small / straightforward).
I am going to trigger a readmapping now.
Already done one for variantcalling and genomenote.

Leave blobtoolkit for you to tru.

muffato commented:
This looks fine in principle but I'm still trying to understand the [BlobToolKit PR](https://github.com/sanger-tol/blobtoolkit/pull/162/files) it made. The indentation change it proposes are not there when I do `nf-core pipelines sync` locally, and although I can reproduce the version 0.7.0→0.7.1 change, I want to check with nf-core folks if that's expected.

muffato commented:
- The fact it picks up the new version is fine. It needs the version and picks it up from the actual `dev` branch. Since the PR is meant to be merged into `dev` anyway, that won't be a conflict.
- I guess the yaml got the indentation wrong because nf-core couldn't run prettier on it as per https://github.com/nf-core/tools/blob/3.2.0/nf_core/pipelines/lint_utils.py#L132-L141 . Maybe a side effect of moving from `pip install .` to `pip install nf-core==...` ? Anyway, can you install prettier in the action ?
- I guess the `ro-crate-metadata.json` update is probably justified. I did tweak the file (and other metadata) on `dev` and nf-core does use those. I'd just accept it.

I'll close the BTK PR for now

muffato commented:
> Because different versions of prettier being picked up? 

For a little while, I thought it was picking up a different configuration file or no configuration file.
In the nf-core setup, prettier is run from the [root of the nf-core/tools](https://github.com/nf-core/tools/tree/3.2.0) checkout which has `.prettierrc.yml` and `.editorconfig` . Why `.editorconfig` matters ? Because it's actually used by prettier:
```
$ prettier --help | grep editorconfig
  --no-editorconfig        Don't take .editorconfig into account when parsing configuration.
```
While here it's called from the root of the pieplines-website checkout

Well, my theory goes in the bin because there [_is_ .editorconfig](https://github.com/sanger-tol/pipelines-website/blob/main/.editorconfig) ... And in fact, locally, if I run `prettier` with the `--no-config` and `--no-editorconfig` options, it actually works perfectly fine.

On the farm I tried prettier 3.3.3 and 3.4.2 and they both work fine.

I'm back to thinking that prettier is not found even though you've installed it. What was wrong with the pip install ? That's how we instal it on the farm in the Conda environments

gq1 commented:
> What was wrong with the pip install ? That's how we instal it on the farm in the Conda environments

pip only install very old version of prettier. All nf-core ci install it via node but nf-core tool get it somehow via pre-commit.

gq1 commented:
> While here it's called from the root of the pieplines-website checkout

In this workflow, the pipelines-website is not checked out. 
So no` .prettierrc.yml` and` .editorconfig`

Here are the difference between pipelins-website and nf-core/tools:

missing:
```
# ignore python and markdown files
[*.py]
indent_style = unset

[**/{CONTRIBUTING,README}.md]
indent_style = unset

[**/Makefile]
indent_style = unset

[tests/pipelines/__snapshots__/*]
charset = unset
end_of_line = unset
insert_final_newline = unset
trim_trailing_whitespace = unset
indent_style = unset
indent_size = unset
```

extra:
```
singleQuote: true
braceStyle: '1tbs'
phpVersion: '7.4'
```

gq1 commented:
@sanger-tolsoft fix linting

muffato commented:
About the `force_pr` option. In nf-core it was meant to force a PR to be open even if there are no changes but they forgot to link the GitHub Action parameter to the `--force_pr` command-line parameter. So essentially they're not using it.

We could delete it, or change it to `make_pr` that controls whether a PR is made at all (`--pull-request`) like I think you did in one of the intermediate commits.

muffato commented:
It now works ! https://github.com/sanger-tol/blobtoolkit/pull/164

gq1 commented:
I deleted PRs and branches for variantcalling and genomenote.
once ready, we can try again. the first time will have lots of changes.
